ui/react-components/components/AppointmentEditor/AppointmentEditor.jsx

```diff
@@ -103,42 +147,89 @@ const AppointmentEditor = props => {
         const startTimeBeforeEndTime = isStartTimeBeforeEndTime(startTime, endTime);
         setPatientError(!isValidPatient);
         setServiceError(!service);
-        setDateError(!startDate);
+        setAppointmentDateError(!appointmentDate);
         setStartTimeError(!startTime);
         setEndTimeError(!endTime);
         setStartTimeBeforeEndTimeError(!startTimeBeforeEndTime);
-        return isValidPatient && service && startDate && startTime && endTime && startTimeBeforeEndTime;
+        return isValidPatient && service && appointmentDate && startTime && endTime && startTimeBeforeEndTime;
     };
 
     const {angularState} = React.useContext(AppContext);
 
+    const isValidRecurringAppointment = () => {
+        const isValidPatient = patient && patient.uuid;
+        const startTimeBeforeEndTime = isStartTimeBeforeEndTime(startTime, endTime);
+        setPatientError(!isValidPatient);
+        setServiceError(!service);
+        setStartTimeError(!startTime);
+        setEndTimeError(!endTime);
+        setStartTimeBeforeEndTimeError(!startTimeBeforeEndTime);
+        setRecurrencePeriodError(!period || period < 1);
+        setEndDateTypeError(!endDateType);
+        if (endDateType) {
+            setEndDateError(endDateType === "On" && !recurringEndDate);
+            setOccurrencesError(endDateType === "After" && (!occurrences || occurrences < 1));
+        }
+        setStartDateError(!startDateType || !recurringStartDate);
+        return isValidPatient && service && startTime && endTime && startTimeBeforeEndTime &&
+            recurrenceType && period && period > 0 && recurringStartDate &&
+            ((endDateType === "On" && recurringEndDate) || (endDateType === "After" && occurrences && occurrences > 0));
+    };
```

Comment:
Please extract it to smaller to add readability

Reply:
Extracted it to a method